### ISSUE TITLE
list_devices() should not stop during first error

### ISSUE THREAD

**Issue by m13253** [joextodd/pysoundio#29]
Test code:
```py
>>> from pysoundio import PySoundIo
>>> pysoundio = PySoundIo()
ALSA lib pcm_dsnoop.c:638:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dsnoop.c:638:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1043:(snd_pcm_dmix_open) The dmix plugin supports only playback stream
ALSA lib pcm_dsnoop.c:575:(snd_pcm_dsnoop_open) The dsnoop plugin supports only capture stream
>>> pysoundio.list_devices()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rpi/.local/lib/python3.7/site-packages/pysoundio/pysoundio.py", line 244, in list_devices
    device = soundio.get_input_device(i)
pysoundio.PySoundIoError: Unable to probe device
```

It seems that some `dsnoop` devices failed to probe. (according to the following `sio_list_devices` output)
I think PySoundIo should skip over the failed devices and mark them as "failed", instead of stop the operation completely.

System version:
```
Linux raspbian 4.14.79-v7+ #1159 SMP Sun Nov 4 17:50:20 GMT 2018 armv7l GNU/Linux

No LSB modules are available.
Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux buster/sid
Release:	testing
Codename:	buster
```

sio_list_devices:
```
ALSA lib pcm_dsnoop.c:638:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dsnoop.c:638:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1043:(snd_pcm_dmix_open) The dmix plugin supports only playback stream
ALSA lib pcm_dsnoop.c:575:(snd_pcm_dsnoop_open) The dsnoop plugin supports only capture stream
--------Input Devices--------

bcm2835 ALSA, bcm2835 ALSA: Direct sample snooping device
  id: dsnoop:CARD=ALSA,DEV=0
  probe error: unable to open device

bcm2835 ALSA, bcm2835 IEC958/HDMI: Direct sample snooping device
  id: dsnoop:CARD=ALSA,DEV=1
  probe error: unable to open device

USB-C to 3.5mm Headphone Jack A, USB Audio: Default Audio Device (default)
  id: default:CARD=A
  channel layouts:
    Mono
  current layout: Mono
  sample rates:
    48000 - 48000
  current sample rate: 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.04266667 sec
  max software latency: 0.34133333 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: Default Audio Device
  id: sysdefault:CARD=A
  channel layouts:
    Mono
  current layout: Mono
  sample rates:
    48000 - 48000
  current sample rate: 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.04266667 sec
  max software latency: 0.34133333 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: Direct sample mixing device
  id: dmix:CARD=A,DEV=0
  probe error: unable to open device

USB-C to 3.5mm Headphone Jack A, USB Audio: Direct sample snooping device
  id: dsnoop:CARD=A,DEV=0
  channel layouts:
    Mono
  current layout: Mono
  sample rates:
    48000 - 48000
  current sample rate: 48000
  formats: signed 16-bit LE
  min software latency: 0.04266667 sec
  max software latency: 0.34133333 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: Direct hardware device without any conversions
  id: hw:CARD=A,DEV=0
  channel layouts:
    Mono
  sample rates:
    44100 - 48000
  formats: signed 16-bit LE
  min software latency: 0.00200000 sec
  max software latency: 10.92266667 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: Hardware device with all software conversions
  id: plughw:CARD=A,DEV=0
  channel layouts:
    Mono
  sample rates:
    44100 - 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.00200000 sec
  max software latency: 10.92266667 sec

USB-C to 3.5mm Headphone Jack A USB Audio (raw)
  id: hw:1,0
  channel layouts:
    Mono
  sample rates:
    44100 - 48000
  formats: signed 16-bit LE
  min software latency: 0.00200000 sec
  max software latency: 10.92266667 sec


--------Output Devices--------

bcm2835 ALSA, bcm2835 ALSA: Default Audio Device
  id: default:CARD=ALSA
  channel layouts:
    Mono
    Stereo
    2.1
    3.0
    3.0 (back)
    3.1
    4.0
    Quad
    Quad (side)
    4.1
    5.0 (back)
    5.0 (side)
    5.1
    5.1 (back)
    6.0 (side)
    6.0 (front)
    Hexagonal
    6.1
    6.1 (back)
    6.1 (front)
    7.0
    7.0 (front)
    7.1
    7.1 (wide)
    7.1 (wide) (back)
    Octagonal
  sample rates:
    8000 - 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.00533333 sec
  max software latency: 1.36533333 sec

bcm2835 ALSA, bcm2835 ALSA: Default Audio Device
  id: sysdefault:CARD=ALSA
  channel layouts:
    Mono
    Stereo
    2.1
    3.0
    3.0 (back)
    3.1
    4.0
    Quad
    Quad (side)
    4.1
    5.0 (back)
    5.0 (side)
    5.1
    5.1 (back)
    6.0 (side)
    6.0 (front)
    Hexagonal
    6.1
    6.1 (back)
    6.1 (front)
    7.0
    7.0 (front)
    7.1
    7.1 (wide)
    7.1 (wide) (back)
    Octagonal
  sample rates:
    8000 - 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.00533333 sec
  max software latency: 1.36533333 sec

bcm2835 ALSA, bcm2835 ALSA: Direct sample mixing device
  id: dmix:CARD=ALSA,DEV=0
  channel layouts:
    Stereo
  sample rates:
    48000 - 48000
  current sample rate: 48000
  formats: signed 16-bit LE
  min software latency: 0.04266667 sec
  max software latency: 0.34133333 sec

bcm2835 ALSA, bcm2835 IEC958/HDMI: Direct sample mixing device
  id: dmix:CARD=ALSA,DEV=1
  channel layouts:
    Stereo
  sample rates:
    48000 - 48000
  current sample rate: 48000
  formats: signed 16-bit LE
  min software latency: 0.04266667 sec
  max software latency: 0.34133333 sec

bcm2835 ALSA, bcm2835 ALSA: Direct hardware device without any conversions
  id: hw:CARD=ALSA,DEV=0
  channel layouts:
    Mono
    Stereo
  sample rates:
    8000 - 48000
  formats: signed 16-bit LE
  min software latency: 0.00533333 sec
  max software latency: 1.36533333 sec

bcm2835 ALSA, bcm2835 IEC958/HDMI: Direct hardware device without any conversions
  id: hw:CARD=ALSA,DEV=1
  channel layouts:
    Stereo
  sample rates:
    44100 - 48000
  formats: signed 16-bit LE
  min software latency: 0.00533333 sec
  max software latency: 0.68266667 sec

bcm2835 ALSA, bcm2835 ALSA: Hardware device with all software conversions
  id: plughw:CARD=ALSA,DEV=0
  channel layouts:
    Mono
    Stereo
    2.1
    3.0
    3.0 (back)
    3.1
    4.0
    Quad
    Quad (side)
    4.1
    5.0 (back)
    5.0 (side)
    5.1
    5.1 (back)
    6.0 (side)
    6.0 (front)
    Hexagonal
    6.1
    6.1 (back)
    6.1 (front)
    7.0
    7.0 (front)
    7.1
    7.1 (wide)
    7.1 (wide) (back)
    Octagonal
  sample rates:
    8000 - 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.00533333 sec
  max software latency: 1.36533333 sec

bcm2835 ALSA, bcm2835 IEC958/HDMI: Hardware device with all software conversions
  id: plughw:CARD=ALSA,DEV=1
  channel layouts:
    Mono
    Stereo
    2.1
    3.0
    3.0 (back)
    3.1
    4.0
    Quad
    Quad (side)
    4.1
    5.0 (back)
    5.0 (side)
    5.1
    5.1 (back)
    6.0 (side)
    6.0 (front)
    Hexagonal
    6.1
    6.1 (back)
    6.1 (front)
    7.0
    7.0 (front)
    7.1
    7.1 (wide)
    7.1 (wide) (back)
    Octagonal
  sample rates:
    44100 - 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.00533333 sec
  max software latency: 0.68266667 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: Default Audio Device (default)
  id: default:CARD=A
  channel layouts:
    Stereo
  current layout: Stereo
  sample rates:
    48000 - 48000
  current sample rate: 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.04266667 sec
  max software latency: 0.34133333 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: Default Audio Device
  id: sysdefault:CARD=A
  channel layouts:
    Stereo
  current layout: Stereo
  sample rates:
    48000 - 48000
  current sample rate: 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.04266667 sec
  max software latency: 0.34133333 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: IEC958 (S/PDIF) Digital Audio Output
  id: iec958:CARD=A,DEV=0
  channel layouts:
    Stereo
  sample rates:
    44100 - 48000
  formats: signed 16-bit LE
  min software latency: 0.00200000 sec
  max software latency: 5.46133333 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: Direct sample mixing device
  id: dmix:CARD=A,DEV=0
  channel layouts:
    Stereo
  current layout: Stereo
  sample rates:
    48000 - 48000
  current sample rate: 48000
  formats: signed 16-bit LE
  min software latency: 0.04266667 sec
  max software latency: 0.34133333 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: Direct sample snooping device
  id: dsnoop:CARD=A,DEV=0
  probe error: unable to open device

USB-C to 3.5mm Headphone Jack A, USB Audio: Direct hardware device without any conversions
  id: hw:CARD=A,DEV=0
  channel layouts:
    Stereo
  sample rates:
    44100 - 48000
  formats: signed 16-bit LE
  min software latency: 0.00200000 sec
  max software latency: 5.46133333 sec

USB-C to 3.5mm Headphone Jack A, USB Audio: Hardware device with all software conversions
  id: plughw:CARD=A,DEV=0
  channel layouts:
    Stereo
  sample rates:
    44100 - 48000
  formats: signed 8-bit, unsigned 8-bit, signed 16-bit LE, signed 16-bit BE, unsigned 16-bit LE, unsigned 16-bit LE, signed 24-bit LE, signed 24-bit BE, unsigned 24-bit LE, unsigned 24-bit BE, signed 32-bit LE, signed 32-bit BE, unsigned 32-bit LE, unsigned 32-bit BE, float 32-bit LE, float 32-bit BE, float 64-bit LE, float 64-bit BE
  min software latency: 0.00200000 sec
  max software latency: 5.46133333 sec

bcm2835 ALSA bcm2835 ALSA (raw)
  id: hw:0,0
  channel layouts:
    Mono
    Stereo
  sample rates:
    8000 - 48000
  formats: signed 16-bit LE
  min software latency: 0.00533333 sec
  max software latency: 1.36533333 sec

bcm2835 ALSA bcm2835 IEC958/HDMI (raw)
  id: hw:0,1
  channel layouts:
    Stereo
  sample rates:
    44100 - 48000
  formats: signed 16-bit LE
  min software latency: 0.00533333 sec
  max software latency: 0.68266667 sec

USB-C to 3.5mm Headphone Jack A USB Audio (raw)
  id: hw:1,0
  channel layouts:
    Stereo
  sample rates:
    44100 - 48000
  formats: signed 16-bit LE
  min software latency: 0.00200000 sec
  max software latency: 5.46133333 sec


27 devices found
```

**Comment by m13253** [comment]
According to [`libsoundio` documentation](http://libsound.io/doc-2.0.0/structSoundIoDevice.html#a129f523c7d43096985c058bc67fe3b6c):

> Ideally probe_error will be SoundIoErrorNone in which case all the fields of the device will be populated. If there is an error code here then information about formats, sample rates, and channel layouts might be missing.

It indicates that, even if probe_error is set, information other than formats, sample rates, and channel layouts may be useful.
Therefore, I think we should not treat probe_error as a hard fatal error.

**Comment by joextodd** [comment]
Resolved in #30
